### PR TITLE
[ENG-7806] prevent tunneled request if no proxy provided

### PR DESCRIPTION
**Pull request by tonatiuh**
What is this PR? 
* Adds a guard condition to prevent sending a request that was supposed to be tunneled but wouldn't be.
  * That could happen when no vgs_proxy_config  was provided, and the `tunnel` option was set to true. 
  * In that case, the request would be sent to the upstream server directly.
  * That's very important to prevent, especially for proxies like VGS that tokenize the information (if the request was not proxied, then the calling server would be the raw data instead of the tokenized data).

### CODE REVIEW

**Comment by tonatiuh** [comment]
Reminder for me: 
* After merging this PR, create the Tilda PR that points to the latest commit of the `alice` branch here.

File: lib/synapse_pay_rest/api/subnets.rb [hunk 1]
```diff
@@ -38,7 +38,8 @@ def get(user_id:, node_id:, subnet_id: nil, **options)
       end.compact
 
       path += '?' + params.join('&') if params.any?
-      tunnel = options[:full_dehydrate] == "yes"
+      tunnel = (options[:full_dehydrate].presence || options["full_dehydrate"].presence) == "yes"
```
Comment (szeidman) [inline lib/synapse_pay_rest/api/subnets.rb:41]: optional potential lint: ```tunnel = options.with_indifferent_access[:full_dehydrate].presence == "yes"```

File: spec/lib/http_client_spec.rb [hunk 2]
```diff
@@ -120,5 +120,26 @@
...
+          ip_address: "127.0.0.1",
+          base_url: "https://uat-api.synapsefi.com/v3.1",
+          fingerprint: "fingerprint_123",
+        )
+
+        client.update_headers(oauth_key: "oauth_key_123")
+        client
+      }
+
+      it "raises erorr to prevent request be sent directly to upstream host" do
```
Comment (szeidman) [inline spec/lib/http_client_spec.rb]: small typo

**Review by szeidman — APPROVED** [review]
LGTM, see comments on a couple tiny things

File: lib/synapse_pay_rest/api/subnets.rb (hunk 1 above)
Comment (tonatiuh) [reply lib/synapse_pay_rest/api/subnets.rb:41]: Thanks. That one would have been actually nice, but `with_indifferent_access` is added by Rails, over here it raises error :(

**Pull request merged**